torch/_refs/__init__.py

```diff
@@ -2038,6 +2039,22 @@ def tensor_split(
         return tuple(splits)
 
 
+def t(a: TensorLikeType):
+    # Currently refs don't support sparse checks
+    # if a.is_sparse:
+    #     sparse_dim = a.sparse_dim()
+    #     dense_dim = a.dense_dim()
+    #     if not (sparse_dim <= 2 and dense_dim == 0):
+    #         raise RuntimeError(
+    #             f"t() expects a tensor with <= 2 sparse and 0 dense dimensions, but got {sparse_dim} sparse and {dense_dim} dense dimensions"
+    #         )
+    if a.ndim > 2:
+        raise RuntimeError(
+            f"t() expects a tensor with <= 2 dimensions, but self is {a.ndim}D"
+        )
+    return torch.transpose(a, 0, 0 if a.ndim < 2 else 1)
```

Comment:
What about 

```
if a.ndim <= 1:
  return prims.view_of(a)
if a.ndim == 2:
  return transpose(a, 0, 1)

raise RuntimeError()
```

?
```

Reply:
Plausibly nicer, but this is how the ATen op is currently implemented: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/TensorShape.cpp#L2503

Reply:
I can add a case to the `transpose` ref to handle when the 2 dims are the same?

Reply:
It's not like we have to implement things exactly as ATen does.

Yeah we can extend the existing short-circuit here instead:

https://github.com/pytorch/pytorch/blob/876c359347faae9367cb390071034e8705bad90e/torch/_refs/__init__.py#L1943

```
if a.ndim <= 1 or dim0 == dim1:
```